src/js/view/controls/components/simple-tooltip.js

```diff
@@ -28,18 +31,32 @@ export function SimpleTooltip(attachToElement, name, text, openCallback, closeCa
                 return;
             }
 
-            removeClass(tooltipElement, 'jw-open');
+            redraw(false);
 
             if (closeCallback) {
                 closeCallback();
             }
         },
         setText(newText) {
-            replaceInnerHtml(textElement, newText);
+            if (newText !== instance.text) {
+                instance.text = newText;
+                instance.dirty = true;
+            }
+            if (instance.opened) {
+                redraw(true);
+            }
         }
     };
 
-    instance.setText(text);
+    const redraw = (opened) => {
+        if (opened && instance.dirty) {
+            replaceInnerHtml(textElement, instance.text);
+            instance.dirty = false;
+        }
+
+        instance.opened = opened;
+        toggleClass(tooltipElement, 'jw-open', open);
```

Comment:
your third arg should be `opened` right? otherwise you're passing a function each time which will fail the check in `toggleClass` for a boolean meaning it will always toggle

Reply:
updated

Reply:
approved